CTFd/utils/user/__init__.py

```diff
@@ -123,7 +123,8 @@ def get_team_attrs(team_id):
 def get_current_user_type(fallback=None):
     if authed():
         user = get_current_user_attrs()
-        return user.type
+        if user and user.type:
```

Comment:
This is a marginal functionality change. If `user.type == None`, the prior functionality returns `None`. Now, we return `fallback`.

Although fallback is defaulted to `None`, calling `get_current_user_type(fallback='XYZ')` for a user with `user.type==None` now returns `'XYZ'` instead of `None`. This will be the case in [3 spots in the code](https://github.com/search?q=repo%3ACTFd%2FCTFd%20get_current_user_type&type=code) where we call `get_current_user_type(fallback="user")`.

If this change is concerning to you, I can change this line to:

`if user and hasattr(user, "type"):`